core/rfb.js

```diff
@@ -1619,12 +1624,22 @@ export default class RFB extends EventTargetMixin {
         this._refreshCursor();
     }
 
+    _shouldShowDot() {
+        if (!this._showDot) { return false; }
+        for (let y = 0; y < this._cursorImage.h; y++) {
+            for (let x = 0; x < this._cursorImage.w; x++) {
+                let idx = 4 * (x + this._cursorImage.w * y) + 3;
+                if (this._cursorImage.rgbaPixels[idx]) { return false; }
```

Comment:
You should probably be prepared for cursors with colourful transparent pixels. (i.e. only check the alpha channel).

Reply:
But I do exactly this! see `4 * ... + 3` formula.